main.go

```diff
@@ -379,6 +379,16 @@ func main() {
 		log.Infof("Wrote subnet file to %s", opts.subnetFile)
 	}
 
+	// back off set node network unavailable for flannel cni bin read subnet.env file read error
+	if opts.setNodeNetworkUnavailable {
+		err = sm.SetNodeNetworkUnavailableFalse(ctx)
+		if err != nil {
+			log.Errorf("Unable to set NodeNetworkUnavailable to False: %v", err)
+		}
+	} else {
+		log.Infoln("Skip setting NodeNetworkUnavailable")
+	}
+
```

Comment:
It probably makes more sense to add it after the backend has already started to make sure everything is ready. Could you add it as part of the condition in line 405? Something like:
```
// Set NetworkUnavailable false in kube as flannel is ready or start monitoring lease for etcd
if opts.kubeSubnetMgr {
// your code
} else {
    err = MonitorLease(ctx, sm, bn, &wg)
    if err == errInterrupted {
    // The lease was "revoked" - shut everything down
        cancel()
     }
  }
```

Reply:
look at new update